src/rpc/blockchain.cpp

```diff
@@ -1926,6 +1929,251 @@ static UniValue savemempool(const JSONRPCRequest& request)
+public:
+    explicit CoinsViewScanReserver() : m_could_reserve(false) {}
+
+    bool reserve() {
+        assert (!m_could_reserve);
+        std::lock_guard<std::mutex> lock(g_utxosetscan);
+        if (g_scan_in_progress) {
+            return false;
+        }
+        g_scan_in_progress = true;
+        m_could_reserve = true;
+        return true;
+    }
+
+    ~CoinsViewScanReserver() {
+        if (m_could_reserve) {
+            std::lock_guard<std::mutex> lock(g_utxosetscan);
+            g_scan_in_progress = false;
+        }
+    }
+};
+
+UniValue scantxoutset(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2)
+        throw std::runtime_error(
+            "scantxoutset <action> {\"pubkeys\": [\"pubkey\",...], \"xpubs\":[{\"xpub\": \"<xpub>\"}]}\n"
+            "\nScans the unspent transaction output set for possible entries that matches common scripts of given public keys.\n"
+            "\nArguments:\n"
+            "1. \"action\"                       (string, required) The action to execute\n"
+            "                                          \"start\" for starting a scan\n"
+            "                                          \"abort\" for aborting the current scan (returns true when abort was successful)\n"
+            "                                          \"status\" for progress report (in %) of the current scan\n"
+            "2. \"options\"                      (object, optional)\n"
+            "      \"pubkeys\":[\"pubkey\",...]    (array of strings, optional) An array of HEX encoded public keys\n"
+            "      \"addresses\":[\"address\",...] (array of strings, optional) An array of bitcoin addresses\n"
+            "      \"xpubs\":                    (array of xpub objects that will be used to derive child keys with the given lookup window after m/0/k and m/1/k scheme)\n"
+            "           [\n"
+            "               {\n"
+            "                \"xpub\":\"<xpub>\",  (base58check encoded extended public key (xpub)\n"
+            "                \"lookupwindow\":   [<startindex>, <stopindex>] (An array with two integers that does define the range of keys that will be deriven for the given xpubs, default is 0 to 1000)\n"
+            "                }\n"
+            "           ]\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"unspents\": [\n"
+            "    {\n"
+            "    \"txid\" : \"transactionid\",     (string) The transaction id\n"
+            "    \"vout\": n,                    (numeric) the vout value\n"
+            "    \"scriptPubKey\" : \"script\",    (string) the script key\n"
+            "    \"amount\" : x.xxx,             (numeric) The total amount in " + CURRENCY_UNIT + " received by the address\n"
+            "    \"height\" : n,                 (numeric) Height of the unspent transaction output\n"
+            "   }\n"
+            "   ,...], \n"
```

Comment:
nit: the spacing of this block looks a bit funny. I think the text should be indented more and the braces for the object should be aligned.